### ISSUE TITLE
Firmware with Bluetooth please

### ISSUE THREAD

**Issue by TedAus** [cvetaevvitaliy/Radiomaster_TX16S#4]
Hi

Thanks for your work on this. I'm really grateful I can now use ACCESS and OTA on my Jumper T16 Pro! 

However, I would also like to use the (Horus) Bluetooth module I got for my Jumper T16 Pro. Would you be kind enough to please compile a version of the firmware with Bluetooth and ACCESS (+ Lua) please? I'm happy to make a contribution for your efforts. 

Many thanks.

**Comment by cvetaevvitaliy** [comment]
Hello!  Thank you for your interest!  Well, I will build FW soon.

**Comment by TedAus** [comment]
Thanks cvetaevvitaliy! Very much appreciated.

**Comment by cvetaevvitaliy** [comment]
Please test this firmware, I do not have T16 and BT module, if BT module is good work I will download it for everyone

[firmware_t16_bt.zip](https://github.com/cvetaevvitaliy/Radiomaster_TX16S/files/5286385/firmware_t16_bt.zip)

**Comment by TedAus** [comment]
Sorry, that did not work because I have a T16 Pro but the firmware is for the non-Pro version.

Thanks.

**Comment by cvetaevvitaliy** [comment]
Do you have such an item on the menu?

![image](https://user-images.githubusercontent.com/26421560/94337588-44d2f300-fff4-11ea-8dd2-3a8eaefa1282.jpeg)

**Comment by TedAus** [comment]
Sorry, I haven't been able to test the Bluetooth yet as I haven't had a chance to solder the module.

**Comment by cvetaevvitaliy** [comment]
And after the firmware did this item appear in the menu?

the one that I threw off above, it is for both Pro and not Pro versions

**Comment by TedAus** [comment]
I do get the Bluetooth line, but don't see the options you have. Probably because the module has not been installed yet.

![PXL_20200926_094728954 MP](https://user-images.githubusercontent.com/71205730/94337973-e4fb3d00-0031-11eb-9ede-24788ae9af97.jpg)

If you look at the screenshot below, Receiver 2 appears three times, instead of Receiver 1, Receiver 2 and Receiver 3. I had this problem previously when I used the non-Pro version.

![PXL_20200926_094745560 MP_2](https://user-images.githubusercontent.com/71205730/94337959-b9785280-0031-11eb-8db1-d9f0ecb99715.jpg)

**Comment by cvetaevvitaliy** [comment]
no, you need to select Bluetooth and turn the encoder, but about the problem above, try to register and bind the receiver

**Comment by cvetaevvitaliy** [comment]
I don't have a Jumper T16 Pro to check it, only Radiomaster TX16S

try to reset the receivers, I see there are 8
Clear receivers to 0

![image](https://user-images.githubusercontent.com/26421560/94338181-fd029a80-fff8-11ea-86f5-9e31ca21707d.jpeg)

**Comment by cvetaevvitaliy** [comment]
added lua support

[firmware.zip](https://github.com/cvetaevvitaliy/Radiomaster_TX16S/files/5286515/firmware.zip)

**Comment by TedAus** [comment]
Ah, yes I now get the Bluetooth options.

![PXL_20200926_103339642 MP](https://user-images.githubusercontent.com/71205730/94338716-cc8e2100-0037-11eb-9347-63c6839b2113.jpg)

I've changed receiver number to 00 but still the same. Receiver 2 appears on all three lines and I can't perform the bind.
HOWEVER, it is interesting that my previously bound receiver was still communicating with the transmitter even though the screen shows no receivers bound!

I had the "Receiver 2, Receiver 2, Receiver 2" problem with [this firmware](https://github.com/cvetaevvitaliy/Radiomaster_TX16S/blob/master/Firmware/Jumper_T16_R9M_ACCESS.bin). But, [this Pro firmware](https://github.com/cvetaevvitaliy/Radiomaster_TX16S/blob/master/Firmware/Jumper_T16_Pro_R9M_ACCESS.bin) fixed it. Does that help you work out what's wrong?

**Comment by cvetaevvitaliy** [comment]
There are no changes in this menu T16, T16Pro, Radiomaster TX16S
I cleaned the project and rebuilt everything - try it
I also checked it in the simulator, I don't see such a problem, try resetting the settings, the EEPROM may be damaged

![Screenshot from 2020-09-26 14-16-33](https://user-images.githubusercontent.com/26421560/94339566-7d2dfd80-0003-11eb-8fbf-63e289145189.png)

[firmware.zip](https://github.com/cvetaevvitaliy/Radiomaster_TX16S/files/5286574/firmware.zip)

**Comment by TedAus** [comment]
Success! I flashed the latest version you provided above and the receiver that I had previously bound with reappeared! And the bluetooth options remain intact. I also tested LUA and for some reason, it seems MUCH faster. In fact, it's almost instant compared to how long it previously took to load each page. Have you optimised the LUA script somehow?

Thanks so much for your help! This is awesome! Look forward to having this integrated into the main stable release.

![PXL_20200926_123627641 MP](https://user-images.githubusercontent.com/71205730/94341060-cdc84980-0049-11eb-811b-5fee03006edc.jpg)
![PXL_20200926_123601462 MP](https://user-images.githubusercontent.com/71205730/94341062-d28cfd80-0049-11eb-8412-dc422557c244.jpg)

**Comment by TedAus** [comment]
I forgot to ask - when power is set to 1W, is that the same as Flex 1W which automatically increases power when signal strength drops? Or does your firmware locks the power at 1W? 

It would be good if you could make the power switch automatically like it does on the Flex firmware.

Thanks again.

**Comment by cvetaevvitaliy** [comment]
it does not block anything, I did not change the OpenTX (this original firmware) code which is responsible for ACCEESS

**Comment by TedAus** [comment]
Ok, thanks! So even though the display doesn't say "Auto", I can assume it actually changes the power output based in signal strength?

**Comment by TedAus** [comment]
Quick update. I've installed the Horus Bluetooth module on my Jumper T16 Pro and managed to get telemetry data using the Telemetry Viewer app. However, it keeps disconnecting after about 10 seconds. Any idea why? My phone also won't pair with it using the 000000 PIN (but can still get the data, albeit only briefly before it disconnects).

**Comment by cvetaevvitaliy** [comment]
I ordered myself a Bluetooth module from AliExpress, I will check and work with it

**Comment by TedAus** [comment]
Nice! Did you get an internal or external model?

**Comment by cvetaevvitaliy** [comment]
https://aliexpress.ru/item/1005001429040420.html?spm=a2g0s.9042311.0.0.2a9433edWaMYx5

**Comment by TedAus** [comment]
Looks the same as what I got. Have to say that it was very easy to solder! 

Note that the module doesn't come with an antenna, so you might have to order it separately.

**Comment by TedAus** [comment]
Hi

I see that 2.3.10 is out but I don't believe ACCESS has been included yet. Could you please make one based on 2.3.10 and include Bluetooth, Lua and ACCESS?

Many thanks.

**Comment by cvetaevvitaliy** [comment]
Hi!

I received my BT module today, solder it in the radio, turned it on and everything works.  I can see it on the network, see the BT MAC address and see the dust MAC address when I connect to it with my laptop

![image](https://user-images.githubusercontent.com/26421560/96923921-9fa81f00-14ba-11eb-8b7c-38b6a5cc462e.jpeg)

**Comment by cvetaevvitaliy** [comment]
![image](https://user-images.githubusercontent.com/26421560/96924046-c49c9200-14ba-11eb-9218-e3da0fc492c0.jpeg)

**Comment by TedAus** [comment]
Nice! Do you find the signal strength to be weak and tends to disconnect after a few minutes?

**Comment by cvetaevvitaliy** [comment]
No, all works good, I don’t have BLE disconnected event

**Comment by jejakmedia** [comment]
> 
> 
> added lua support
> 
> [firmware.zip](https://github.com/cvetaevvitaliy/Radiomaster_TX16S/files/5286515/firmware.zip)

 I don't want to change acces on tx16s, is there any firmware to activate bluetooth only?
